src/Microsoft.Framework.Runtime/PatternsCollectionHelper.cs

```diff
@@ -53,7 +58,12 @@ private static IEnumerable<string> CreateCollection(string projectDirectory, par
             {
                 if (Path.IsPathRooted(pattern))
                 {
-                    throw new InvalidOperationException(string.Format("Patten {0} is a rooted path, which is not supported.", pattern));
+                    throw new InvalidOperationException($"Pattern {pattern} is a rooted path, which is not supported.");
+                }
+
+                if (literalPath && pattern.Contains('*'))
+                {
+                    throw new InvalidOperationException($"Pattern {pattern} is not expected to contain wildcard characters.");
```

Comment:
I think a message like this is a little clearer: 

```
The '{propertyName}' property cannot contain wildcard characters, use '{wildcardPropertyName}' to specify wildcard patterns.
```

For example:

```
The 'compileFiles' property cannot contain wildcard characters, use 'compile' to specify wildcard patterns.
```

It does mean you need to pass down `propertyName` and add a `wildcardPropertyName` parameter, but I think the clearer error message is worth it.